isthmus/src/main/java/io/substrait/isthmus/expression/ExpressionRexConverter.java

```diff
@@ -0,0 +1,289 @@
+  }
+
+  @Override
+  public RexNode visit(Expression.TimestampTZLiteral expr) throws RuntimeException {
+    return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory, expr.getType()));
+  }
+
+  @Override
+  public RexNode visit(Expression.IntervalYearLiteral expr) throws RuntimeException {
+    // return rexBuilder.makeLiteral(expr.value(), TypeConverter.convert(typeFactory,
+    // expr.getType()));
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.IntervalDayLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.UUIDLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.FixedCharLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.VarCharLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.FixedBinaryLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.DecimalLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.MapLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.ListLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
+        this.getClass().getCanonicalName()
+            + " does not support "
+            + expr.getClass().getCanonicalName());
+  }
+
+  @Override
+  public RexNode visit(Expression.StructLiteral expr) throws RuntimeException {
+    throw new UnsupportedOperationException(
```

Comment:
As mentioned in #27, let's use an abstract visitor here with fallback for the failure cases.